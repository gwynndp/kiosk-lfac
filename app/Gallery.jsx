import React from 'react';
import VideoCard from './VideoCard';

export default function Gallery({ videos }) {
  return <div>{videos.map(video => <VideoCard key={video.name} name={video.name} src={video.src} />)}</div>;
}
