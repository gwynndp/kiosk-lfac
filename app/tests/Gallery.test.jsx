import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Gallery from '../Gallery';

test('should render a Gallery', () => {
  const props = {
    videos: [
      {
        id: 4,
        category: 'people',
        name: 'Ashley Sanchez',
        src: 'videos/Ashley-Sanchez.mp4',
        img: 'images/videos/Ashley-Sanchez.jpg'
      },
      {
        id: 5,
        category: 'people',
        name: 'Jackie Rose Wilson',
        src: 'videos/Jackie-Rose-Wilson.mp4',
        img: 'images/videos/Jackie-Rose-Wilson.jpg'
      },
      {
        id: 6,
        category: 'people',
        name: 'Jude Gibson Byers',
        src: 'videos/Jude-Gibson-Byers.mp4',
        img: 'images/videos/Jude-Gibson-Byers.jpg'
      },
      {
        id: 7,
        category: 'fires',
        name: 'Kathryn Hecht',
        src: 'videos/Kathryn-Hecht.mp4',
        img: 'images/videos/Kathryn-Hecht.jpg'
      },
      {
        id: 8,
        category: 'fires',
        name: 'Leanne Fogg',
        src: 'videos/Leanne-Fogg.mp4',
        img: 'images/videos/Leanne-Fogg.jpg'
      }
    ],
    showVideoView: false,
    handleVideoCardClick: () => {}
  };
  const tree = renderer
    .create(
      <Router>
        <Gallery videos={props.videos} handleVideoCardClick={props.handleVideoCardClick} />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

// should render the correct # of GalleryCards based on props.videos
