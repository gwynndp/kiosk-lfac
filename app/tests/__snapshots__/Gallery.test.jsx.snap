// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a Gallery 1`] = `
<div
  className="sc-AykKF gqeRDE"
  id="gallery"
>
  <a
    href="/video"
    onClick={[Function]}
    style={
      Object {
        "color": "black",
        "textDecoration": "none",
      }
    }
  >
    <div
      className="sc-AykKE jDVYoJ"
      onClick={[Function]}
    >
      <h1
        className="sc-AykKC flIKka"
      >
        Ashley Sanchez
      </h1>
      <img
        alt="Video titled Ashley Sanchez"
        className="sc-AykKD kcIegN"
        src="images/videos/Ashley-Sanchez.jpg"
      />
    </div>
  </a>
  <a
    href="/video"
    onClick={[Function]}
    style={
      Object {
        "color": "black",
        "textDecoration": "none",
      }
    }
  >
    <div
      className="sc-AykKE jDVYoJ"
      onClick={[Function]}
    >
      <h1
        className="sc-AykKC flIKka"
      >
        Jackie Rose Wilson
      </h1>
      <img
        alt="Video titled Jackie Rose Wilson"
        className="sc-AykKD kcIegN"
        src="images/videos/Jackie-Rose-Wilson.jpg"
      />
    </div>
  </a>
  <a
    href="/video"
    onClick={[Function]}
    style={
      Object {
        "color": "black",
        "textDecoration": "none",
      }
    }
  >
    <div
      className="sc-AykKE jDVYoJ"
      onClick={[Function]}
    >
      <h1
        className="sc-AykKC flIKka"
      >
        Jude Gibson Byers
      </h1>
      <img
        alt="Video titled Jude Gibson Byers"
        className="sc-AykKD kcIegN"
        src="images/videos/Jude-Gibson-Byers.jpg"
      />
    </div>
  </a>
  <a
    href="/video"
    onClick={[Function]}
    style={
      Object {
        "color": "black",
        "textDecoration": "none",
      }
    }
  >
    <div
      className="sc-AykKE jDVYoJ"
      onClick={[Function]}
    >
      <h1
        className="sc-AykKC flIKka"
      >
        Kathryn Hecht
      </h1>
      <img
        alt="Video titled Kathryn Hecht"
        className="sc-AykKD kcIegN"
        src="images/videos/Kathryn-Hecht.jpg"
      />
    </div>
  </a>
  <a
    href="/video"
    onClick={[Function]}
    style={
      Object {
        "color": "black",
        "textDecoration": "none",
      }
    }
  >
    <div
      className="sc-AykKE jDVYoJ"
      onClick={[Function]}
    >
      <h1
        className="sc-AykKC flIKka"
      >
        Leanne Fogg
      </h1>
      <img
        alt="Video titled Leanne Fogg"
        className="sc-AykKD kcIegN"
        src="images/videos/Leanne-Fogg.jpg"
      />
    </div>
  </a>
  <div
    className="spacer"
    style={
      Object {
        "flexGrow": "1",
      }
    }
  />
</div>
`;
